core/ibft.go

```diff
@@ -861,24 +865,30 @@ func (i *IBFT) runCommit(ctx context.Context) error {
 	defer i.messages.Unsubscribe(sub.ID)
 
 	for {
+		if i.handleCommit(view) {
+			i.signalRoundDone(ctx)
+
+			return
+		}
+
+		//	quorum not reached, retry
 		select {
 		case <-ctx.Done():
 			// Stop signal received, exit
-			return errTimeoutExpired
+			return
 		case <-sub.SubCh:
-			if !i.handleCommit(view) {
```

Comment:
Is there reason why the code is moved above the select?

Reply:
Same as aboce